website/source/docs/internals/providers/resource-data.html.md

```diff
@@ -0,0 +1,180 @@
+---
+layout: "docs"
+page_title: "Using ResourceData"
+sidebar_current: "docs-internals-provider-guide-resource-data"
+description: |-
+  Details about the ubiquitous ResourceData type and all its options.
+---
+
+# Using `ResourceData`
+
+When working with the `Create`, `Read`, `Update`, `Destroy`, and `Exists`
+methods on your Provider, it's almost impossible to not run into
+`ResourceData`. It's a type that is used all over, so it's important to
+understand it. But `ResourceData` can also be tricky, because it's not
+abstracting a single logical concept in Terraform.
+
+## How to Conceptualise `ResourceData`
+
+The most useful way to think of `ResourceData` is not as a placeholder for any
```

Comment:
I read this sentence 3 times and I'm still not sure what is being said. Could you rephrase or break into separate sentences please?